deepedit_prostate/main.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import os
+import glob
+
+from lib import Deepgrow, MyStrategy, MyTrain, Segmentation, dynunet_strides_kernels
+from monai.networks.nets.dynunet_v1 import DynUNetV1
+
+from monailabel.interfaces import MONAILabelApp
+from monailabel.utils.activelearning import Random
+
+logger = logging.getLogger(__name__)
+
+
+class MyApp(MONAILabelApp):
+    def __init__(self, app_dir, studies):
+        strides, kernels = dynunet_strides_kernels.dynUNet_strides_kernels(patch_size=(192, 192, 32), target_spacing=(1.0, 1.0, 3.0))
+        self.network = DynUNetV1(
+            spatial_dims=3,
+            in_channels=3,
+            out_channels=1,
+            kernel_size=kernels,
+            strides=strides,
+            upsample_kernel_size=strides[1: ],
+            norm_name="instance",
+            deep_supervision=False,
+            res_block=True,
+        )
+
+        self.model_dir = os.path.join(app_dir, "model")
+        self.pretrained_model = os.path.join(self.model_dir, "pretrained.pt")
+        self.final_model = os.path.join(self.model_dir, "model.pt")
+
+        self.download(
+            [
+                (
+                    self.pretrained_model,
+                    "https://github.com/Project-MONAI/MONAILabel/releases/download/data/deepedit_prostate.pt",
```

Comment:
This link not exists

Reply:
@diazandr3s can you update the link after you have saved the .pt I sent you?

Reply:
Thanks! Here is the link

https://github.com/diazandr3s/MONAILabel-Apps/releases/download/pretrained/deepedit_prostate.pt

Reply:
Please make sure it works as you'd expect

Reply:
download failed - does it need to be renamed pretrained.pt? See trace:

[2021-07-15 15:23:17,676] [INFO] (uvicorn.error) - Started server process [17464]
[2021-07-15 15:23:17,678] [INFO] (uvicorn.error) - Waiting for application startup.
[2021-07-15 15:23:17,679] [INFO] (monailabel.utils.others.app_utils) - Initializing App from: C:\Users\pmeht\Documents\MONAILabel-Apps\deepedit_prostate; studies: C:\Users\pmeht\Documents\monai_label\data\PROSTATEx\imagesTr
[2021-07-15 15:23:18,384] [INFO] (monailabel.interfaces.app) - Downloading resource: C:\Users\pmeht\Documents\MONAILabel-Apps\deepedit_prostate\model\pretrained.pt from https://github.com/Project-MONAI/MONAILabel/releases/download/data/deepedit_prostate.pt
pretrained.pt: 0.00B [00:01, ?B/s]
Download failed from https://github.com/Project-MONAI/MONAILabel/releases/download/data/deepedit_prostate.pt to C:\Users\pmeht\AppData\Local\Temp\tmpobastpdg\pretrained.pt.

Reply:
I think you copy the wrong link. Please copy exactly this one: https://github.com/diazandr3s/MONAILabel-Apps/releases/download/pretrained/deepedit_prostate.pt

Reply:
Yes, you are correct. Now works. A model folder is created with pretrained.pt within it.